### PR TITLE
Add methods to read EC2 data and download from S3

### PR DESCRIPTION
**Pull request by zachmargolis**
This is mostly a port of this un-merged code in https://github.com/18F/identity-idp/pull/1557

My goal is that we could update the part in `environment.rb` to look like this:

```ruby
root = File.expand_path('../../', __FILE__)

LoginGov::Hostdata.in_datacenter do |hostdata|
  hostdata.s3.download_configs(
    '/%{env}/v1/idp/application.yml' => File.join(root, 'config/application_s3.yml'),
    '/%{env}/v1/idp/database.yml'    => File.join(root, 'config/database_s3.yml')
  )
end
```

### CODE REVIEW

File: lib/login_gov/hostdata.rb [hunk 1]
```diff
@@ -28,6 +30,23 @@ def self.in_datacenter?
...
+      yield self if in_datacenter?
+    end
+
+    # @return [S3]
+    def self.s3
+      ec2 = LoginGov::Hostdata::EC2.load
+
+      LoginGov::Hostdata::S3.new(
+        env: env,
+        bucket: "login-gov-app-secrets-#{ec2.region}-#{ec2.account_id}",
```
Comment (zachmargolis) [inline lib/login_gov/hostdata.rb]: @brodygov -- is this ok to hardcode here? it simplifies calling code a bunch, and I think our plan is to have like `/idp` directories for the IDP etc etc, so it should be fine?

File: lib/login_gov/hostdata/s3.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      def download_configs(configs)
+        configs.each do |s3_path, local_path|
+          download_config(s3_path, local_path)
+        end
+      end
+
+      private
+
+      def s3_client
+        @s3_client ||= Aws::S3::Client.new
```
Comment (brodygov) [inline lib/login_gov/hostdata/s3.rb]: Do we need region here? Or do instances automatically pick the region that they're currently running in?

File: lib/login_gov/hostdata.rb (hunk 1 above)
Comment (brodygov) [reply lib/login_gov/hostdata.rb]: Yeah it seems fine to hardcode this.

The names are generated as `login-gov.app-secrets.<ACCOUNT_ID>-<REGION>`.

File: lib/login_gov/hostdata/s3.rb (hunk 2 above)
Comment (zachmargolis) [reply lib/login_gov/hostdata/s3.rb]: My understanding is the AWS SDK automatically picks the region it's running in via env vars and magic stuff? But it's easy to make that a param if you're not sure?

Comment (zachmargolis) [reply lib/login_gov/hostdata/s3.rb]: Pulled in region in 4c6af40ec72917a81d7f6efa5ecafd18d085e55f

**Pull request merged**